lib/Doctrine/DBAL/Migrations/Migration.php

```diff
@@ -95,14 +95,18 @@ public function writeSqlFile($path, $to = null)
 
         $this->outputWriter->write(sprintf("-- Migrating from %s to %s\n", $from, $to));
 
-        $sqlWriter = new SqlFileWriter(
+        return $this->createSqlFileWriter($path)
+                    ->write($sql, $direction);
+    }
+
+    protected function createSqlFileWriter(string $path): SqlFileWriter
```

Comment:
@lcobucci urgh. I am really reluctant to introduce a new `protected` method that will be abused till eternity :S

A huge test change would be more acceptable than this, IMO...